### ISSUE TITLE
You shouldn't rely on this as a CSRF protection mechanism

### ISSUE THREAD

**Issue by defendtheworld** [dxa4481/CORS#1]
We wrote a blog post about this as well. These mechanisms are not viable for CSRF protection, and a Flash bypass still exists that can be leveraged from Chrome. https://medium.com/@longtermsec/chrome-just-hardened-the-navigator-beacon-api-against-cross-site-request-forgery-csrf-690239ccccf

**Comment by dxa4481** [comment]
Yeah, that's what I put the disclaimer at the top for. That said, I appreciate that this safety net exists, for new developers that don't know about CSRF, but I wouldn't count on it as a sole line of defense given how quick the chromium devs where to suggest throwing it away.

**Comment by defendtheworld** [comment]
I don't completely disagree. However, browser vendors have been sloppy in this area, and it's not a proper safety net for CSRF. I think it's misleading to call it that. Developers will read this and create applications which are vulnerable to CSRF until all the other vectors are closed down.

**Comment by dxa4481** [comment]
I think they've been sloppy because they don't realize the whole picture, which is all I've tried to do here; paint the picture. I wrote this blog post more for the browser developers than I did individual developers.
I reached out to Adobe asking them to fix their redirect issue, and used this blog post (and yours) as a reference. 
Developers are going to continue to write applications vulnerable to CSRF because CSRF is confusing, unintuitive, and insecure by default. They'll do this regardless of the existence of this blog post.
